pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -98,12 +97,12 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 	configClusterInfo := config.getCluster()
 
 	clientConfig := &restclient.Config{}
-	clientConfig.Host = configClusterInfo.Server
-	if u, err := url.ParseRequestURI(clientConfig.Host); err == nil && u.Opaque == "" && len(u.Path) > 1 {
-		u.RawQuery = ""
-		u.Fragment = ""
-		clientConfig.Host = u.String()
+
+	clientConfig.AlternateHosts = configClusterInfo.Servers
+	if configClusterInfo.Server != "" {
```

Comment:
Hrm - this is not the same behavior.  Did you move this stripping of query and fragment somewhere else?